### PR TITLE
Facebook auth

### PR DESCRIPTION
**Pull request by AhmedAdelIbrahim**
@myriameayman @renadshaaban @som3a123 @ebrahim-elgaml @MariamTalaat @mahatalaat 
PLEASE REVIEW ME

### CODE REVIEW

**Comment by mahatalaat** [comment]
@AhmedAdelIbrahim  please add the documentation so I can be able to review it

File: app/controllers/facebook_controller.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    profile = @graph.get_object("me")
+
+    friends = @graph.get_connections("me", "friends")
+    @user.name = env["omniauth.auth"].info.name
+    @user.provider = env["omniauth.auth"].provider
+    @user.uid = env["omniauth.auth"].uid
+    @user.fb_email = env["omniauth.auth"].info.email
+    @user.faceboook_profileimage=env["omniauth.auth"].info.image
+    @user.oauth_token = env["omniauth.auth"].credentials.token
+    @user.oauth_expires_at = Time.at(auth.credentials.expires_at)
```
Comment (myriameayman) [inline app/controllers/facebook_controller.rb]: @AhmedAdelIbrahim something not working here

**Comment by myriameayman** [comment]
@AhmedAdelIbrahim not working sorry :(

**Comment by AhmedAdelIbrahim** [comment]
@myriameayman I fixed the error , check again if anything bad happened

**Comment by AhmedAdelIbrahim** [comment]
@renadshaaban @som3a123 please review my uml diagram (Facebook-uml)

**Comment by myriameayman** [comment]
:+1: scenario review

**Comment by AhmedAdelIbrahim** [comment]
@myriameayman Ok , Thank you

**Comment by AhmedAdelIbrahim** [comment]
@mahatalaat Ok , I will add it now

**Comment by renadshaaban** [comment]
@AhmedAdelIbrahim  I'm done , good job (Y)

**Comment by AhmedAdelIbrahim** [comment]
@mahatalaat  Please review the documentation of my code , i finished it .

**Comment by AhmedAdelIbrahim** [comment]
@ebrahim-elgaml Please review my code

**Comment by AhmedAdelIbrahim** [comment]
@renadshaaban Thank you very much for your time

**Comment by mahatalaat** [comment]
@AhmedAdelIbrahim  in Facebook_controller 

Line 4 : add dot at the end of the line 
line 29: leave space between # and @ and add a dot at the end 
line 61 : add dot

**Comment by AhmedAdelIbrahim** [comment]
@mahatalaat I have changed them , Please check again

**Comment by mahatalaat** [comment]
@AhmedAdelIbrahim  done

**Comment by AhmedAdelIbrahim** [comment]
@myriameayman Please review me

**Comment by AhmedAdelIbrahim** [comment]
@mahatalaat Thank you very much

**Comment by myriameayman** [comment]
@AhmedAdelIbrahim  u didn't check the uniqueness of fb email, fb image and fb token

**Comment by ebrahim-elgaml** [comment]
@AhmedAdelIbrahim  good job code review done

**Comment by myriameayman** [comment]
@AhmedAdelIbrahim ur test are not valid :( 
test "check validations facebookMail is unique" u actually don't check for the uniqueness, the test is not valid because user doesn't contain fb token 
test "check validations facebookOauth-token is unique" plz create 2 users in the test with the same token the same with test "check validations facebook-uid is unique"

**Comment by AhmedAdelIbrahim** [comment]
@ebrahim-elgaml Thank you very much

**Comment by AhmedAdelIbrahim** [comment]
@myriameayman I have checked them and you are right , I fixed them please check them again and Thank you

**Pull request merged**